### PR TITLE
Small improvements to DefInfo – Part 1

### PR DESCRIPTION
**Pull request by chexee**
- Formats code blocks in docstcings
- Fixes “Sign in” button spacing
- Improved empty state for no refs or examples

### CODE REVIEW

**Comment by beyang** [comment]
Screenshot highlighting visual changes would be great!

File: app/web_modules/sourcegraph/components/styles/_typography.css [hunk 1]
```diff
@@ -47,10 +47,9 @@
 	line-height: 1.45;
 }
 
-@value code-normal-line-height 22px; /* used by JavaScript */
 .code-normal {
 	font-size: code-font-size-2;
-	line-height: code-normal-line-height;
+	line-height: 1.45;
```
Comment (sqs) [inline app/web_modules/sourcegraph/components/styles/_typography.css]: @chexee This is used in FileMargin.js to position the def box in the blob view. It needs to be accessible to JavaScript as a pixel value for the computations to work. Check the top of FileMargin.js, where it uses this value.

File: app/web_modules/sourcegraph/def/index.js [hunk 2]
```diff
@@ -94,6 +94,6 @@ export type RefLocationsKey = {
 
 export type ExamplesKey = {
 	repo: string;
-	commitID: string;
-	def: string;
+	commitID: ?string[];
```
Comment (sqs) [inline app/web_modules/sourcegraph/def/index.js]: Why an array (`[]`)?

File: app/web_modules/sourcegraph/def/index.js [hunk 3]
```diff
@@ -94,6 +94,6 @@ export type RefLocationsKey = {
...
+	def: ?string[];
```
Comment (sqs) [inline app/web_modules/sourcegraph/def/index.js]: Same - why an array?

File: app/web_modules/sourcegraph/def/index.js [hunk 4]
```diff
@@ -85,7 +85,7 @@ export function encodeDefPath(path: string): string {
 
 export type RefLocationsKey = {
 	repo: string;
-	commitID: string;
+	commitID: ?string[];
```
Comment (sqs) [inline app/web_modules/sourcegraph/def/index.js]: Same - why an array?

File: app/web_modules/sourcegraph/components/styles/_typography.css (hunk 1 above)
Comment (sqs) [inline app/web_modules/sourcegraph/components/styles/_typography.css]: If you have a better way to do this, just make sure that the def box is aligned with the selected def on the blob view after your changes.

**Comment by chexee** [comment]
Empty state logged in:
screenshot 2016-07-08 10 56 32

Empty state logged out:
screenshot 2016-07-08 10 56 59

File: app/web_modules/sourcegraph/components/styles/_typography.css (hunk 1 above)
Comment (chexee) [inline app/web_modules/sourcegraph/components/styles/_typography.css]: Ah, I'll take a look – ideally we can calculate the vertical position of the highlighted line after it's rendered independently of the line height of the page.

File: app/web_modules/sourcegraph/def/index.js (hunk 2 above)
Comment (chexee) [inline app/web_modules/sourcegraph/def/index.js]: Ah, I misunderstood the flow syntax here. It should allow for both `string` and `null`, which I think would actually just be `?string`

**Comment by sqs** [comment]
"joining with GitHub" not Github in the screenshot

File: app/web_modules/sourcegraph/blob/BlobMain.js [hunk 5]
```diff
@@ -180,6 +188,7 @@ export default class BlobMain extends Container {
 					<Blob
 						repo={this.state.repo}
 						rev={this.state.rev}
+						ref={(c) => { this.setState({selectionStartLine: c && c.refs && c.refs.startLineComponent ? c.refs.startLineComponent : null}); }}
```
Comment (beyang) [inline app/web_modules/sourcegraph/blob/BlobMain.js]: Adding parens to make explicit the order of operations here would make this more readable. I'm actually not sure of the order myself. Is it

`(c && c.refs && c.refs.startLineComponent) ? c.refs.startLineComponent : null`

or

`c && c.refs && (c.refs.startLineComponent ? c.refs.startLineComponent: null)`

**Comment by beyang** [comment]
1 minor comment, otherwise LGTM :ship:

File: app/web_modules/sourcegraph/blob/BlobMain.js (hunk 5 above)
Comment (chexee) [inline app/web_modules/sourcegraph/blob/BlobMain.js]: The first one, so parens sounds like a  great idea :)

**Pull request merged**